src/board.rs

```diff
@@ -70,6 +70,46 @@ pub fn get_piece_from_char(ch: char) -> Piece {
     }
 }
 
+#[derive(Debug)]
+enum Orientation {
+    Rank,
+    File,
+    Both,
+}
+
+#[derive(Debug)]
+enum InvalidOffsetError {
+    LessThanZero(Orientation, Offset),
```

Comment:
I'm not sure I understand the reasoning behind adding a separate `InvalidOffsetError`.

When do you think it will be useful to an end user of `board` to know that, for example, the offset they gave is somehow an invalid offset, over and above an `InvalidSquare` error? What extra information could a consumer use from the `InvalidOffsetError`?

Reply:
My thought on that was that a square can never have coordinates that were less than zero so it made sense to have a separate type for that since the InvalidSquareError might be used in a context where the other error type wouldn't be a valid error for that case